tool/tbot/config/destination_directory.go

```diff
@@ -58,23 +61,143 @@ func (dd *DestinationDirectory) CheckAndSetDefaults() error {
 		return trace.BadParameter("destination path must not be empty")
 	}
 
+	secureSupported, err := botfs.HasSecureWriteSupport()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	aclsSupported, err := botfs.HasACLSupport()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	switch dd.Symlinks {
+	case "":
+		if secureSupported {
+			// We expect Openat2 to be available, so try to use it by default.
+			dd.Symlinks = botfs.SymlinksSecure
+		} else {
+			// TrySecure will print a warning on fallback.
+			dd.Symlinks = botfs.SymlinksTrySecure
+		}
+	case botfs.SymlinksInsecure, botfs.SymlinksTrySecure:
+		// valid
+	case botfs.SymlinksSecure:
+		if !secureSupported {
+			return trace.BadParameter("symlink mode %q not supported on this system", dd.Symlinks)
+		}
+	default:
+		return trace.BadParameter("invalid symlinks mode: %q", dd.Symlinks)
+	}
+
+	switch dd.ACLs {
+	case "":
+		if aclsSupported {
+			// Unlike openat2(), we can't ever depend on ACLs being available.
+			// We'll only ever try to use them, end users can opt-in to a hard
+			// ACL check if they wish.
+			dd.ACLs = botfs.ACLTry
+		} else {
+			// if aclsSupported == false here, we know it will never work, so
+			// don't bother trying.
+			dd.ACLs = botfs.ACLOff
+		}
+	case botfs.ACLOff, botfs.ACLTry:
+		// valid
+	case botfs.ACLOn:
+		if !aclsSupported {
+			return trace.BadParameter("acls mode %q not supported on this system", dd.ACLs)
+		}
+	default:
+		return trace.BadParameter("invalid acls mode: %q", dd.ACLs)
+	}
+
 	return nil
 }
 
-func (dd *DestinationDirectory) Write(name string, data []byte, modeHint destination.ModeHint) error {
-	// TODO: honor modeHint?
-	if err := os.WriteFile(filepath.Join(dd.Path, name), data, 0600); err != nil {
+func (dd *DestinationDirectory) Init() error {
+	// Create the directory if needed.
+	stat, err := os.Stat(dd.Path)
+	if trace.IsNotFound(err) {
+		err = os.MkdirAll(dd.Path, botfs.DefaultDirMode)
+		log.Infof("Created directory %q", dd.Path)
```

Comment:
maybe log after the error check so you don't log if something failed?

Reply:
Wow, how'd I miss that one? Thanks!